File: src/main/java/field/movement/fragment/JumpMoveFragment.java
package field.movement.fragment;

import util.packet.PacketReader;
import util.packet.PacketWriter;

import java.awt.*;

public class JumpMoveFragment extends ActionMoveFragment {

    Point vposition;

    public JumpMoveFragment(byte movePathAttribute, PacketReader packetReader) {
        super(movePathAttribute, packetReader);
    }

    @Override
    public void decodeData(PacketReader packet) {
        vposition = packet.readPoint();

        super.decodeData(packet);
    }

    @Override
    public void encodeData(PacketWriter packet) {
        packet.writePosition(vposition);

        super.encodeData(packet);
    }
}

File: src/main/java/field/movement/fragment/JumpMoveFragment.kt
package field.movement.fragment

import util.packet.PacketReader
import util.packet.PacketWriter
import java.awt.Point

class JumpMoveFragment(movePathAttribute: Byte, packetReader: PacketReader) : ActionMoveFragment(movePathAttribute, packetReader) {

    lateinit var vposition: Point

    override fun decodeData(packet: PacketReader) {
        vposition = packet.readPoint()
        super.decodeData(packet)
    }

    override fun encodeData(packet: PacketWriter) {
        packet.writePosition(vposition)
        super.encodeData(packet)
    }
}